### PR TITLE
[14.0][IMP] stock_request: Add domain to some fields to prevent deprecate methods with domain return

### PR DESCRIPTION
**Pull request by victoralmau**
Add domain to some fields to prevent deprecate methods with domain return

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT41080

### CODE REVIEW

File: stock_request/models/stock_request_abstract.py [hunk 1]
```diff
@@ -37,12 +37,17 @@ def _compute_product_qty(self):
 
     name = fields.Char("Name", copy=False, required=True, readonly=True, default="/")
     warehouse_id = fields.Many2one(
-        "stock.warehouse", "Warehouse", ondelete="cascade", required=True
+        comodel_name="stock.warehouse",
+        string="Warehouse",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```
Comment (pedrobaeza) [inline stock_request/models/stock_request_abstract.py]: I think this should go with new multi-company features (like check_company=True).

File: stock_request/models/stock_request_abstract.py [hunk 2]
```diff
@@ -37,12 +37,17 @@ def _compute_product_qty(self):
...
+        required=True,
     )
     location_id = fields.Many2one(
-        "stock.location",
-        "Location",
-        domain=[("usage", "in", ["internal", "transit"])],
+        comodel_name="stock.location",
+        string="Location",
+        domain="allow_virtual_location and [(1, '=', 1)] or "
+        "not allow_virtual_location and [('usage', 'in', ['internal', 'transit'])]",
```
Comment (pedrobaeza) [inline stock_request/models/stock_request_abstract.py]:
```suggestion
        domain="not allow_virtual_location and [('usage', 'in', ['internal', 'transit'])] or [(1, '=', 1)]",
```

**Comment by victoralmau** [comment]
Changes done.

File: stock_request/models/stock_request_abstract.py [hunk 3]
```diff
@@ -37,12 +37,17 @@ def _compute_product_qty(self):
...
+        required=True,
     )
     location_id = fields.Many2one(
-        "stock.location",
-        "Location",
-        domain=[("usage", "in", ["internal", "transit"])],
+        comodel_name="stock.location",
+        string="Location",
+        domain="not allow_virtual_location and "
+        "[('usage', 'in', ['internal', 'transit'])] or [(1, '=', 1)]",
```
Comment (pedrobaeza) [inline stock_request/models/stock_request_abstract.py]: Not sure if it should be empty domain instead:
```suggestion
        "[('usage', 'in', ['internal', 'transit'])] or []",
```

**Review by pedrobaeza — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**